angularSTEP/src/app/profile-card/profile-card.component.spec.ts

```diff
@@ -0,0 +1,36 @@
+import { async, ComponentFixture, TestBed, inject, tick, fakeAsync } from '@angular/core/testing';
+import { DebugElement, Component } from '@angular/core';
+
+import { By } from '@angular/platform-browser';
+
+import { ProfileCardComponent } from './profile-card.component';
+
+describe('ProfileCardComponent', () => {
+    let component: ProfileCardComponent;
+    let fixture: ComponentFixture<ProfileCardComponent>;
+    let de: DebugElement;
+
+    beforeEach(async(() =>{
+        TestBed.configureTestingModule({
+            declarations: [ ProfileCardComponent ],
+
+        })
+        .compileComponents();
+    }));
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(ProfileCardComponent);
+        component = fixture.componentInstance;
+        de = fixture.debugElement;
+    });
+
+    it('should create', () => {
+        expect(component).toBeTruthy();
+    });
+
+    it('should be a string', () =>{
```

Comment:
It seems like this is intended to be the main reason for this change so it'd be nice to make sure it has good coverage of `profile-card.component.ts`! While it's a relatively simple component, and this does ensure the fields get initialized, I see that there's logic to `shortenDisplayName()` so could that also be tested in this change?

Reply:
Overall LGTM but I'll save approval for after the `shortenDisplayName()` test is added here. You may need to merge main branch in first?

Reply:
Does this actually check initialization, or does it just test the typing system? It looks like all of these properties are typed without an "!" or "|".

Also, these look like they are all properties of UserData, instead of properties of the ProfileCardComponent. How does that work?